metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -224,40 +224,12 @@ object MetalsEnrichments extends DecorateAsJava with DecorateAsScala {
       }
     }
 
-    /**
-     * Symlink-aware version of Files.createDirectories()
-     */
-    def createDirectories(): AbsolutePath = {
-      val nioPath = path.toNIO
-      findExistingParent(nioPath) match {
-        case Some(parent) =>
-          parent.relativize(nioPath).iterator.asScala.foldLeft(parent) {
-            (parent, dirName) =>
-              val dirPath = parent.resolve(dirName)
-              // This check guards against symlinks
-              // isDirectory follows symlinks but the check inside `createDirectory` doesn't
-              if (!Files.isDirectory(dirPath)) {
-                Files.createDirectory(dirPath)
-              }
-              dirPath
-          }
-        // we couldn't find a parent
-        case None => throw new FileSystemException(path.toString)
+    def createDirectories(): Unit = {
+      if (!Files.isSymbolicLink(path.toNIO)) {
```

Comment:
If `path` is a symlink to a regular file, this will silently fail and do nothing.

I would use

```scala
if (!Files.isDirectory(path.toNIO)) {
  Files.createDirectories(path.toNIO)
}
```

instead

Reply:
To elaborate, using `isDirectory` in that scenario will allow `createDirectories` to fail with a meaningful error, instead of silently ignoring the issue.

Reply:
Good catch! I updated the logic to run `createDirectories` on the dealiased path. If the symlink points to a file it should fail with the correct message in that case.